l10n_ch_qr_bill_scan/wizard/account_invoice_import.py

```diff
@@ -0,0 +1,159 @@
+# Copyright 2020 Camptocamp
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+import logging
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+from odoo.modules.module import get_resource_path
+
+from ..tools import QR
+
+logger = logging.getLogger(__name__)
+
+try:
+    import cv2
+    import numpy as np
+    import pdf2image
+    from pyzbar.pyzbar import ZBarSymbol, decode
+except ImportError:
+    # Necessary libraries to decode QR from pdf are
+    # not installed
+    logger.warning(
```

Comment:
Even though the package was installed successfully, we still require [these libraries](https://github.com/sonhd91/l10n-switzerland/pull/17/commits/b12b3649743dfb7b397e039780e169016426a467#diff-eb0aab1a9ec98c524c9a1268e5496b0104dde25bc901ae4c6e7df278ccc98cf5R5-R6) for it to function it properly. If we don't include them, test case would fail, could we add these lib to test-requirement file?

Reply:
can you add this in the manifest:
```
    "external_dependencies": {
        "deb": ["libzbar0"],
        "python": ["pyzbar", "pdf2image", "opencv-python", "numpy"],
    },
```
libzbar0 is needed for pyzbar indeed
but I don't think that `poppler-utils libzbar-dev zbar-tools` are still needed

Reply:
ok my bad, `poppler-utils` needed then

Reply:
- I think the good practice is to never raise an error at import level, because it would prevent the whole odoo instance from loading: we just log the error and only the concerned module will fail
- anyway, now that `external_dependencies` has been defined, dep availability will be checked by odoo at install

Reply:
I tried add this to manifest file, but it seems like Travis doesn't install it

```
  "external_dependencies": {
        "deb": ["libzbar0", "poppler-utils"],
        "python": ["pyzbar", "pdf2image", "opencv-python", "numpy"],
    },
```